src/osd/PG.cc

```diff
@@ -1428,15 +1428,28 @@ bool PG::choose_acting(pg_shard_t &auth_log_shard_id,
     }
   }
 
-  // We go incomplete if below min_size for ec_pools since backfill
+  // We go incomplete if below data_chunk count for ec_pools since backfill
   // does not currently maintain rollbackability
   // Otherwise, we will go "peered", but not "active"
-  if (num_want_acting < pool.info.min_size &&
-      (pool.info.ec_pool() ||
-       !cct->_conf->osd_allow_recovery_below_min_size)) {
-    want_acting.clear();
-    dout(10) << "choose_acting failed, below min size" << dendl;
-    return false;
+  if (num_want_acting < pool.info.min_size) { 
+    bool ret = true;
+    if (pool.info.ec_pool()) {
+      auto& ecp = get_osdmap()->get_erasure_code_profile(pool.info.erasure_code_profile);
+      auto pk = ecp.find("k");
+      assert(pk != ecp.end());
+      unsigned int k = atoi(pk->second.c_str());
```

Comment:
I don't think this is quite right, since some of the erasure codes are quite odd.

Luckily, we already have a programmatic interface to tell if we have enough shards to recover a PG. It's the next check with the recoverable_predicate!

I think we just need to move this block below that check and then we don't need to do this check ourselves.